From 8cae8be25afecc584660838ca43aa2afda2b3c62 Mon Sep 17 00:00:00 2001
From: Gerd Hoffmann <kraxel@redhat.com>
Date: Mon, 22 Nov 2010 16:54:01 -0200
Subject: [PATCH 3/7] Add defines for PCI IDs.

RH-Author: Gerd Hoffmann <kraxel@redhat.com>
Message-id: <1290444845-14392-4-git-send-email-kraxel@redhat.com>
Patchwork-id: 13789
O-Subject: [RHEL-6.1 vgabios PATCH 3/7] Add defines for PCI IDs.
Bugzilla: 654639
RH-Acked-by: Amit Shah <amit.shah@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>

This patch allows to set PCI vendor and device IDs using defines
(PCI_VID and PCI_DID).  Use it for vgabios.bin.
(cherry picked from commit bb44e12f7b1b7ccd040c58ddb0e8cdaf2f9f5a67)

Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
---
 Makefile  |    4 ++--
 vbe.c     |    6 +++++-
 vgabios.c |    5 +++++
 3 files changed, 12 insertions(+), 3 deletions(-)

Signed-off-by: Eduardo Habkost <ehabkost@raisama.net>
---
 Makefile  |    4 ++--
 vbe.c     |    6 +++++-
 vgabios.c |    5 +++++
 3 files changed, 12 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index c3c744c..d440b93 100644
--- a/Makefile
+++ b/Makefile
@@ -31,8 +31,8 @@ VGA_FILES := vgabios.c vgabios.h vgafonts.h vgatables.h
 VBE_FILES := vbe.h vbe.c vbetables.h
 
 # build flags
-vgabios.bin              : VGAFLAGS := -DVBE
-vgabios.debug.bin        : VGAFLAGS := -DVBE -DDEBUG
+vgabios.bin              : VGAFLAGS := -DVBE -DPCI_VID=0x1234
+vgabios.debug.bin        : VGAFLAGS := -DVBE -DPCI_VID=0x1234 -DDEBUG
 vgabios-cirrus.bin       : VGAFLAGS := -DCIRRUS -DPCIBIOS 
 vgabios-cirrus.debug.bin : VGAFLAGS := -DCIRRUS -DPCIBIOS -DCIRRUS_DEBUG
 
diff --git a/vbe.c b/vbe.c
index 3bd4d43..af9de66 100644
--- a/vbe.c
+++ b/vbe.c
@@ -910,7 +910,11 @@ Bit16u *AX;Bit16u CX; Bit16u ES;Bit16u DI;
                 if (using_lfb) {
                   info.NumberOfBanks = 1;
                 }
-                lfb_addr = pci_get_lfb_addr(0x1234); // experimental vendor
+#ifdef PCI_VID
+                lfb_addr = pci_get_lfb_addr(PCI_VID);
+#else
+                lfb_addr = 0;
+#endif
                 if (lfb_addr > 0) {
                   info.PhysBasePtr = ((Bit32u)lfb_addr << 16);
                 }
diff --git a/vgabios.c b/vgabios.c
index fbc3588..b3a5cec 100644
--- a/vgabios.c
+++ b/vgabios.c
@@ -210,8 +210,13 @@ vgabios_pci_data:
 .word 0x1013
 .word 0x00b8 // CLGD5446
 #else
+#ifdef PCI_VID
+.word PCI_VID
+.word PCI_DID
+#else
 #error "Unknown PCI vendor and device id"
 #endif
+#endif
 .word 0 // reserved
 .word 0x18 // dlen
 .byte 0 // revision
-- 
1.7.3.2

